filter/bytes_filter.go

```diff
@@ -0,0 +1,92 @@
+package filter
+
+import (
+	"bytes"
+
+	"github.com/xichen2020/eventdb/values/iterator"
+	xbytes "github.com/xichen2020/eventdb/x/bytes"
+)
+
+// BytesFilter matches against bytes values.
+type BytesFilter interface {
+	// Match returns true if the given value is considered a match.
+	Match(v iterator.Bytes) bool
+}
+
+type bytesToBytesFilterFn func([]byte) bytesFilterFn
+
+type bytesFilterFn func(v []byte) bool
+
+func (fn bytesFilterFn) Match(v iterator.Bytes) bool { return fn(v.Data) }
+
+func equalsBytesBytes(rhs []byte) bytesFilterFn {
+	return func(lhs []byte) bool {
+		return bytes.Equal(lhs, rhs)
+	}
+}
+
+func notEqualsBytesBytes(rhs []byte) bytesFilterFn {
```

Comment:
nit: rename to `notEqualsBytesFn`

Reply:
See earlier comment on naming convention.